spec/index.bs

```diff
@@ -412,6 +412,34 @@ for the [=usage scope=] of this render pass.
 
 Issue(gpuweb/gpuweb#514): Document read-only states for depth views.
 
+## Synchronization ## {#programming-model-synchronization}
+
+For each [=subresource=] of a [=physical resource=], the usage is tracked on the [=Queue timeline=].
+
+Timeline could be split into a sequence of [=usage scope=]:
+within each scope the usage of the [=subresource=] stays unchanged,
+and the implementation transitions the [=subresource=] to the new usage
+at the [=usage scope=] boundaries.
+
+This specification defines the following [=usage scopes=]:
+  1. an individual command on a {{GPUCommandEncoder}}, such as {{GPUCommandEncoder/copyBufferToTexture()|GPUCommandEncoder.copyBufferToTexture}}.
+  2. an individual command on a {{GPUComputePassEncoder}}, such as {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)|GPUProgrammablePassEncoder.setBindGroup}}.
+  3. the whole {{GPURenderPassEncoder}}.
+
+The [=usage scopes=] are fully contained within command buffers,
+therefore submitting a sequence of command buffers,
+which have been validated for correct resource usage during recording on the [=Content timeline=],
+results in the resource usage also being valid at the [=Queue timeline=].
+
+The implementation does the <dfn dfn>usage scope validation</dfn> by composing
+the union of all usage flags of the [=subresources=] used by the [=usage scope=].
+The {{GPUValidationError}} is generated in the current scope with appropriate error message
+if the union contains a [=mutating usage=] combined with any other usage.
+
+Note: an implementation is likely to validate the usage scopes earlier
+than at {{GPUQueue/submit()|GPUQueue.submit}} time, where the [=Queue timeline=] is defined.
+For example, it can check if all the bindings inside a single {{GPUBindGroup}} make a valid usage scope
+that would be considered every time {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)}} is called.
```

Comment:
I don't understand what this is saying; we can talk about it on the editors call.

Reply:
resolved: We don't need this particular bit. But we should move the validation from submit() to finish().